lib/LANraragi/Utils/I18NInitializer.pm

```diff
@@ -0,0 +1,58 @@
+package LANraragi::Utils::I18NInitializer;
+
+use strict;
+use warnings;
+use utf8;
+use Encode;
+use LANraragi::Utils::I18N;
+
+sub initialize {
+    my ($app) = @_;
+
+    $app->helper(
+        lh => sub {
+            my ($c, $key, @args) = @_;
+
+            my $accept_language = $c->req->headers->accept_language;
+            my $lang = 'en';
+
+            if ($accept_language) {
+                ($lang) = split /,/, $accept_language;
+                $lang =~ s/-.*//;
+                $c->log->debug("Detected language: $lang");
+            }
+
+            my $handle = LANraragi::Utils::I18N->get_handle($lang);
+
+            unless ($handle) {
+                $c->log->debug("No handle for language: $lang");
+                return $key;
```

Comment:
This means that if I'm using an accept_language header that's not in the lexicons, it'll return the localisation key as-is, right?  

Wouldn't this break the few keys we have that require to go through `maketext` for argument processing? (Like `"Version %1 %2"`)  

It might be better to just set the handle to `LANraragi::Utils::I18N->get_handle("en")` here and let the rest of the sub proceed as normal.

Reply:
This won't lead to an issue since `Locale::Maketext`, when it doesn't find a corresponding translation file, will automatically default to the English language file. It only returns the key if the corresponding key is not found in the currently matched language file.

However, I believe this is an area that can be improved. I've made some modifications: essentially, if a translator makes a typo in the key value or encounters other problems, it will automatically use the corresponding `msgstr` from the English file.